torch/csrc/api/include/torch/optim/serialize.h

```diff
@@ -207,6 +207,23 @@ void serialize(
   }
 }
 
+template <typename T>
+c10::List<T> deque_to_list(const std::deque<T>& dq) {
+  c10::List<Tensor> list;
+  list.reserve(dq.size());
+  for (const auto& e : dq) {
+    list.push_back(e);
+  }
+  return list;
+}
+
+template <typename T>
+void list_to_deque(const c10::List<T>& list, std::deque<T>& dq) {
```

Comment:
we should probably make this function return `std::deque<T>` instead of taking in a `std::deque<T>` non-const ref, to be more similar to the signature of `deque_to_list`

Reply:
we need to pass by reference because we want to update dq :D

Reply:
it would prevent a copy as the values would be directly written at name() in memory

Reply:
If we return a `std::deque<T>` from this function it won't do a copy either, because of NRVO (https://www.fluentcpp.com/2016/11/28/return-value-optimizations/) and the fact that name() can take an rvalue as input